rttest/src/rttest.cpp

```diff
@@ -58,17 +60,28 @@ class rttest_sample_buffer
       this->buffer_size = new_buffer_size;
       this->latency_samples = static_cast<int *>(
         std::malloc(new_buffer_size * sizeof(int)));
-      // TODO(dirk-thomas) this must check if the memory was allocated, same below
+      if (!this->latency_samples) {
+        fprintf(stderr, "Failed to allocate latency samples buffer\n");
+        abort();
```

Comment:
I once designed this library with the hope that it could be linked to C executables, hence I avoided using exceptions.